### DISCUSSION TITLE
Xcframework doesn't contain dSYMs

### DISCUSSION THREAD

**Discussion by Cyberbeni** [airbnb/lottie-ios#2282] [General]
In case of a crash while using the lottie-spm package, it would be useful to be able to symbolicate the stack trace.

Here is the script that I use for my library: https://github.com/Cyberbeni/TypedNotificationCenter/blob/master/create-xcframework.sh

PS:
SWIFT_SERIALIZE_DEBUGGING_OPTIONS=NO is required due to this issue: https://github.com/facebook/facebook-ios-sdk/issues/1628
-IDEBuildLocationStyle=Unique doesn't matter unless you are running the script on a machine where you set up workspace relative derived data location as described here: https://github.com/fastlane/fastlane/pull/20184

**Comment by calda** [comment]
@Cyberbeni, which part of your script causes dSYMs to be included in the XCFramework?

Our XCFramework build script is [here](https://github.com/airbnb/lottie-ios/blob/master/Rakefile#L65). Any suggestions as to what changes we could make to include dSYMs? PRs are welcome if you're interested!

**Reply by Cyberbeni** [reply]
Passing the archive path and the framework name is equivalent to passing in the framework path and dsym path (and bitcode path if it wasn't deprecated)

```
CREATE_XCFRAMEWORK_ARGUMENTS+=( "-archive" "$PWD/$ARCHIVE_PATH" )
CREATE_XCFRAMEWORK_ARGUMENTS+=( "-framework" "TypedNotificationCenter.framework" )
```

```
FRAMEWORK_PATH=$(find "$ARCHIVE_PATH" -name "TypedNotificationCenter.framework" -print -quit)
if [ -n "$FRAMEWORK_PATH" ]; then
	CREATE_XCFRAMEWORK_ARGUMENTS+=( "-framework" "$PWD/$FRAMEWORK_PATH" )
fi
DSYM_PATH=$(find "$ARCHIVE_PATH" -name "TypedNotificationCenter.framework.dSYM" -print -quit)
if [ -n "$DSYM_PATH" ]; then
	CREATE_XCFRAMEWORK_ARGUMENTS+=( "-debug-symbols" "$PWD/$DSYM_PATH" )
fi
```
